`utoipa-swagger-ui` broken by latest `zip` release

It looks like the latest zip release busts the `utoipa-swagger-ui` build. This appears to be the offending [commit](https://github.com/zip-rs/zip2/pull/277), which changed `InvalidArchive`. Here's a minimal reproducer:

```
[package]
name = "test"
version = "0.1.0"
edition = "2024"

[dependencies]
utoipa = "5.3.1"
utoipa-swagger-ui = { version = "9.0.0" }
```

```
$ cargo build
...
Compiling utoipa-swagger-ui v9.0.0
error[E0308]: mismatched types
   --> /Users/adamsimpson/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/utoipa-swagger-ui-9.0.0/build.rs:105:49
    |
105 |                 .ok_or(ZipError::InvalidArchive("invalid path file"))?;
    |                        ------------------------ ^^^^^^^^^^^^^^^^^^^ expected `Cow<'_, str>`, found `&str`
    |                        |
    |                        arguments to this enum variant are incorrect
    |
    = note:   expected enum `Cow<'static, str>`
            found reference `&'static str`
note: tuple variant defined here
   --> /Users/adamsimpson/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zip-2.5.0/src/result.rs:23:5
    |
23  |     InvalidArchive(Cow<'static, str>),
    |     ^^^^^^^^^^^^^^
help: try wrapping the expression in `std::borrow::Cow::Borrowed`
    |
105 |                 .ok_or(ZipError::InvalidArchive(std::borrow::Cow::Borrowed("invalid path file")))?;
    |                                                 +++++++++++++++++++++++++++                   +

For more information about this error, try `rustc --explain E0308`.
```

Pinning `zip` to the previous release works as expected:
```
[package]
name = "test"
version = "0.1.0"
edition = "2024"

[dependencies]
utoipa = "5.3.1"
utoipa-swagger-ui = { version = "9.0.0" }

[build-dependencies]
zip = "=2.4.2"
```

--- a/utoipa-swagger-ui/build.rs
+++ b/utoipa-swagger-ui/build.rs
@@ -1,5 +1,4 @@
 use std::{
-    borrow::Cow,
     env,
     error::Error,
     fs::{self, File},
@@ -103,7 +102,7 @@ impl SwaggerZip {
             let mut file = self.by_index(index)?;
             let filepath = file
                 .enclosed_name()
-                .ok_or(ZipError::InvalidArchive(Cow::Borrowed("invalid path file")))?;
+                .ok_or(ZipError::InvalidArchive("invalid path file".into()))?;
 
             if index == 0 {
                 zip_top_level_folder = filepath

--- a/utoipa-swagger-ui/build.rs
+++ b/utoipa-swagger-ui/build.rs
@@ -1,4 +1,5 @@
 use std::{
+    borrow::Cow,
     env,
     error::Error,
     fs::{self, File},
@@ -102,7 +103,7 @@ impl SwaggerZip {
             let mut file = self.by_index(index)?;
             let filepath = file
                 .enclosed_name()
-                .ok_or(ZipError::InvalidArchive("invalid path file"))?;
+                .ok_or(ZipError::InvalidArchive(Cow::Borrowed("invalid path file")))?;
 
             if index == 0 {
                 zip_top_level_folder = filepath

